### PR TITLE
Add a helper method for publishing status events in http rpc transport

### PR DESCRIPTION
**Pull request by rudigiesler**
We need to add a helper method to the base http rpc transport that makes use of the component added in #1004.

### CODE REVIEW

**Comment by rudigiesler** [comment]
Ready for review.

File: vumi/transports/httprpc/httprpc.py [hunk 1]
```diff
@@ -193,6 +195,16 @@ def setup_transport(self):
             ],
             self.web_port)
 
+        self.status_detect = StatusEdgeDetector()
+
+    def add_status(self, **kw):
+        '''Publishes a status if it is not a repeat of the previously
+        published status.'''
+        status = TransportStatus(**kw)
+        if self.status_detect.check_status(status):
```
Comment (hodgestar) [inline vumi/transports/httprpc/httprpc.py]: Given how `publish_status` accepts `**kw`, I wonder if the edge detector shouldn't work the same way? Or at least have the option of being used that way.

File: vumi/transports/httprpc/httprpc.py [hunk 2]
```diff
@@ -193,6 +195,16 @@ def setup_transport(self):
...
+            return self.publish_status(**kw)
+        return succeed(None)
```
Comment (hodgestar) [inline vumi/transports/httprpc/httprpc.py]: Is return a deferred here just future proofing?

**Comment by hodgestar** [comment]
Left two comments, but looks good.

**Comment by hodgestar** [comment]
On other thought -- does `add_status` belong on the base transport perhaps?

File: vumi/transports/httprpc/httprpc.py (hunk 1 above)
Comment (rudigiesler) [inline vumi/transports/httprpc/httprpc.py]: Yes I didn't realise that `publish_status` worked that way, I thought that it accepted a `TransportStatus`. I'll change the edge detector to work that way.

File: vumi/transports/httprpc/httprpc.py (hunk 2 above)
Comment (rudigiesler) [inline vumi/transports/httprpc/httprpc.py]: It's not strictly necessary, I just think it's nice to always return a deferred.

**Comment by rudigiesler** [comment]
I don't know if it belongs in the base transport, since not all transports need it. Things like the SMPP transport only sends a status on certain events, so it won't have the problem of sending the same status over and over again. It could maybe go in the base transport if we find that other transports need it, but so far it's just the HTTP RPC transport that needs it

File: vumi/transports/httprpc/httprpc.py (hunk 2 above)
Comment (hodgestar) [inline vumi/transports/httprpc/httprpc.py]: Agreed.

**Comment by rudigiesler** [comment]
Changed the way that the edge detector takes in arguments. Ready for review again.

**Comment by justinvdm** [comment]
:+1: from me.

**Pull request merged**